### PR TITLE
Update to Node 16 and npm versions

### PR DESCRIPTION
**Pull request by william-msft**
- [ ] The purpose of this PR is explained in this message or in an issue. If an issue please include a reference as https://devdiv.visualstudio.com/DevDiv/_boards/board/t/Oryx/Stories/?workitem=1448431
- [ ] Tests are included and/or updated for code changes.
- [ ] Proper license headers are included in each file.

### CODE REVIEW

File: doc/supportedPlatformVerions.md [hunk 1]
```diff
@@ -170,6 +170,10 @@
 - 15.2.0
 - 15.2.1
 - 15.3.0
+- 16.5.0
```
Comment (gpcastro) [inline doc/supportedPlatformVerions.md:173]: do we support all these? I see only 16.13 now

File: build/constants.yaml [hunk 2]
```diff
@@ -189,19 +189,19 @@
...
     node8-version: 8.17.0
     node10-version: 10.23.0
     node12-version: 12.22.6
     node14-version: 14.17.6
-    node16-version: 16.8.0
+    node16-version: 16.13.1
     node-app-insights-sdk-version: 1.8.7
     pm2-version: 4.5.6
-    npm-version: 6.14.15
+    npm-version: 8.1.2
```
Comment (gpcastro) [inline build/constants.yaml:204]: where is this version used? are we sure it is compatible with the node versions we use it with?

**Review by gpcastro — APPROVED** [review]

**Review by qianz2 — APPROVED** [review]

File: build/constants.yaml (hunk 2 above)
Comment (william-msft) [reply build/constants.yaml:204]: @gpcastro npm 8.1.2 is used during the builds:
image
This is built successfully and will run oryx-nightly to make sure E2E tests are passing successfully

File: doc/supportedPlatformVerions.md (hunk 1 above)
Comment (william-msft) [reply doc/supportedPlatformVerions.md:173]: Removed. Those were added by our script which automatically populates versions

**Pull request merged**